TESTING.md

```diff
@@ -0,0 +1,53 @@
+<!-- TOC -->
+- [Testing](#testing)
+    - [Code Coverage Collecting](#code-coverage-collecting)
+    - [Code Coverage Reporting](#code-coverage-reporting)
+<!-- TOC -->
+
+## Testing
+
+### Code Coverage Collecting
+Code coverage is a measurement of how many lines of code are executed under the test suites.
+
+#### OpenSearch plugin
+
+##### Gradle project
+1. Add [JaCoCo Gradle plugin](https://docs.gradle.org/current/userguide/jacoco_plugin.html) to `build.gradle` in your repository. [example](https://github.com/opensearch-project/sql/blob/opensearch-1.0.0.0-rc1/build.gradle#L48)
+2. Set Jacoco to export coverage report in `xml` format. [example](https://github.com/opensearch-project/sql/blob/opensearch-1.0.0.0-rc1/build.gradle#L101-L105)
+3. Call gradle task "jacocoTestReport" to generate report (`./gradlew jacocoTestReport`) after running your tests. (Or add a `dependsOn` to make "jacocoTestReport" task depends on your test tasks. [example](https://github.com/opensearch-project/index-management/blob/opensearch-1.0.0.0-beta1/build-tools/coverage.gradle))
+
+##### Maven project
+Since OpenSearch uses Gradle as the build tool, we don't recommend using Maven.
+Code coverage for a Maven project can be generated by [JaCoCo Maven plugin](https://www.eclemma.org/jacoco/trunk/doc/maven.html), and we also got an [example](https://github.com/opensearch-project/security/blob/v1.0.0.0-rc1/pom.xml#L636).
+
+#### OpenSearch Dashboards plugin
+
+#### Limitation
```

Comment:
Can we also add the limitations of JaCoCo plugin?
As far as I know, it doesn't support PowerMock and Integration tests(can you double check on this, as this was the case for AD).

Reply:
Thanks for your idea! I will add them 😄.

Reply:
Added the useful tips as far as I know for using JaCoCo.